com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs

```diff
@@ -157,19 +157,12 @@ private void Render(ScriptableRenderContext context, CommandBuffer cmd, ref Rend
             var activeDebugHandler = GetActiveDebugHandler(renderingData);
             if (activeDebugHandler != null)
             {
-                foreach (DebugRenderSetup debugRenderSetup in activeDebugHandler.CreateDebugRenderSetupEnumerable(context, cmd))
-                {
-                    DrawingSettings debugDrawSettings = debugRenderSetup.CreateDrawingSettings(ref renderingData, drawSettings);
-
-                    if (debugRenderSetup.GetRenderStateBlock(out RenderStateBlock renderStateBlock))
-                    {
-                        context.DrawRenderers(renderingData.cullResults, ref debugDrawSettings, ref filterSettings, ref renderStateBlock);
-                    }
-                    else
+                RenderStateBlock renderStateBlock = new RenderStateBlock();
+                activeDebugHandler.DrawWithDebugRenderState(context, cmd, ref renderingData, ref drawSettings, ref filterSettings, ref renderStateBlock,
+                    (ref RenderingData data, ref DrawingSettings ds, ref FilteringSettings fs, ref RenderStateBlock rsb) =>
                     {
-                        context.DrawRenderers(renderingData.cullResults, ref debugDrawSettings, ref filterSettings);
-                    }
-                }
+                        context.DrawRenderers(data.cullResults, ref ds, ref fs, ref rsb);
+                    });
```

Comment:
I think this is nicer and more concise now, but I'm not totally happy with it... there are restrictions how ref paramters can be used with lambdas:
1) seems like ref parameters in the must be named explicitly as ref (so I have to say `ref DrawingSettings ds` instead of just `ds`)
2) more annoyingly, ref parameters cannot be captured from outer scope and used inside the lambda. For this reason, I have to pass renderingData and filterSettings to `DrawWithDebugRenderState` even though they are not modified. If I'm missing how this could be done more cleanly, let me know!